### PR TITLE
Add e2e test for automatic orphaned volume cleanup

### PR DESCRIPTION
**Pull request by tnozicka**
**Description of your changes:**
Adds e2e coverage for automatic orphaned volume cleanup.

### CODE REVIEW

File: pkg/controller/helpers/mergepatch.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import (
+	"fmt"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func GenerateMergePatch(original runtime.Object, modified runtime.Object) ([]byte, error) {
+	if original == nil {
```
Comment (zimnx) [inline pkg/controller/helpers/mergepatch.go:12]: these are interface objects so plain nil comparison doesn't work. I think you can skip these checks

File: test/e2e/set/scyllacluster/scyllacluster_pv.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		sc, err = f.ScyllaClient().ScyllaV1().ScyllaClusters(f.Namespace()).Patch(
+			ctx,
+			sc.Name,
+			types.MergePatchType,
+			[]byte(`{"metadata": {"annotations": {"foo": "bar"} } }`),
+			metav1.PatchOptions{},
+		)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for the PVC to be replaced")
```
Comment (zimnx) [inline test/e2e/set/scyllacluster/scyllacluster_pv.go:94]: entire scylla node should be replaced, we could check that Pod UUID is different

Comment (tnozicka) [reply test/e2e/set/scyllacluster/scyllacluster_pv.go:94]: although PVC has the finalizer waiting for pod deletion

Comment (zimnx) [reply test/e2e/set/scyllacluster/scyllacluster_pv.go:94]: right, this implies pod deletion

**Review by zimnx — DISMISSED** [review]
LGTM, one nit

**Review by zimnx — APPROVED** [review]

**Pull request merged**